templates/master/00-master/_base/files/usr-local-bin-openshift-recovery-tools-sh.yaml

```diff
@@ -188,30 +187,46 @@ contents:
     EOF
     }
 
-    # add member cluster
+    # add member to cluster
     etcd_member_add() {
       echo "Updating etcd membership.."
       if [ -d "$ETCD_DATA_DIR" ]; then
         echo "Removing etcd data_dir $ETCD_DATA_DIR.."
         rm -rf $ETCD_DATA_DIR
       fi
 
-      RESPONSE=$(env ETCDCTL_API=3 $ETCDCTL --cert $ASSET_DIR/backup/etcd-client.crt --key $ASSET_DIR/backup/etcd-client.key --cacert $ASSET_DIR/backup/etcd-ca-bundle.crt \
-        --endpoints ${RECOVERY_SERVER_IP}:2379 member add $ETCD_NAME --peer-urls=https://${ETCD_DNS_NAME}:2380)
-
-       if [ $? -eq 0 ]; then
-         echo "$RESPONSE"
-         APPEND_CONF=$(echo "$RESPONSE" | sed -e '1,2d')
-         echo -e "\n\n#[recover]\n$APPEND_CONF" >> $ETCD_CONFIG
-       else
-         echo "$RESPONSE"
-         exit 1
-       fi
+      RESPONSE=$($ETCDCTL_WITH_TLS --endpoints ${RECOVERY_SERVER_IP}:2379 member add $ETCD_NAME --peer-urls=https://${ETCD_DNS_NAME}:2380)
+      if [ $? -eq 0 ]; then
+        echo "$RESPONSE"
+        APPEND_CONF=$(echo "$RESPONSE" | sed -e '1,2d')
+        echo -e "\n\n#[recover]\n$APPEND_CONF" >> $ETCD_CONFIG
+      else
+        echo "$RESPONSE"
+        exit 1
```

Comment:
This is a generic comment on the entire set of utility functions: I don't like utility function doing `exit` instead of `return`. I prefer control going back to the caller, instead of exiting, but that is for some other time.

Reply:
exit is quick but for retry etc return is better I agree.